developer/kmcompx/tests/kmcompxtest.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * Tiny frame console app to compile a .kmx from a .kmn
+ */
+
+#include <windows.h>
+#include <stdio.h>
+#include "kmcompx.h"
+#include <iostream>
+#include <vector>
+#include <string>
+#include <sstream>
+using namespace std;
+
+vector < int > error_vec;
+
+int WINAPI msgproc(int line, KMX_DWORD dwMsgCode, char* szText)
+{
+  error_vec.push_back(dwMsgCode);
+  printf("line %d  error %x  %s\n", line, (unsigned int)dwMsgCode, szText);
+	return 1;
+}
+
+int main(int argc, char *argv[])
+{
+	if(argc < 3)
+	{
+		puts("Usage: kmcompxtest infile.kmn outfile.kmx");
+		return 1;
+	}
+
+	for(char *p = argv[1]; *p; p++) {
+		if(*p == '/') *p = '\\';
```

Comment:
I would rather not do this:
We would need to check on/change several places because we use #if defined(_WIN32) || defined(_WIN64) a lot  eg. when finding the separator in pszfile ( in CompileKeyboardFile() ): 


```
#if defined(_WIN32) || defined(_WIN64)
  if (p = strrchr(pszInfile, '\\'))
#else
  if (p = strrchr(pszInfile, '/'))
#endif
  {
    strncpy_s(CompileDir, _countof(CompileDir), pszInfile, (INT_PTR)(p - pszInfile + 1));  // I3481
    CompileDir[(INT_PTR)(p - pszInfile + 1)] = 0;
  }
  else
    CompileDir[0] = 0;
```

If we then use CompileDir (in ImportBitmapFile for example) we may end up using CompileDir without path => not finding the file

Reply:
My preference is to shift this problem sideways by moving the file system interactions out of the kmcompx code and into the caller -- see my comment https://github.com/keymanapp/keyman/pull/6980#pullrequestreview-1113028991, point 3.